### PR TITLE
Delete child segments

### LINKED ISSUE #34: Update waveform when a header from structure is deleted
### Description

When a header with timespans is deleted from the structure, the waveform does not reflect the change.

### Done Looks Like

When a header with timespans is deleted the segments in the waveform corresponding to the timespans should be deleted.

### PR DESCRIPTION
**Pull request by Dananji**
This fixes #34 
@adamjarling I created a new branch with all the changes from master and added the changes from feature branch without doing a rebase on master.

### CODE REVIEW

**Comment by adamjarling** [comment]
@Dananji Hey, here's a quick Git trick for being able to avoid the extra Git commit for removing the `console.log`.  Say you realized you want to change something small after the commit.  You can do:

`git reset HEAD~`  

And that will put all your last commit's code changes back into 'unstaged' territory.   Make whatever change you want.  Then stage all your code, including your latest change:

`git add .`

Then re-commit without having to update the commit message, or create a new commit.

`git commit -C ORIG_HEAD`

Basically this just patches a commit you have locally.

If patching a commit you've already pushed to the remote branch, you can do everything above, and then do a force push back up to the remote:

`git push -f origin/your-branch-name-here`

Someone taught me this at a previous job and I do it all the time.
